### PR TITLE
add check for "parentPolicy is not NULL" before using

### PR DESCRIPTION
**Pull request by red1452**
Pointer to the parentPolicy variable is set from pointer
"parentrel->rd_cdbpolicy". This variable  is checked before using at other
places at the transformDistributedBy function in case of the IsBinaryUpgrade
flag is false. At the last "if" the parentPolicy variable is used without
checking. The parentPolicy variable can be NULL for binary upgrade. For binary
upgrade this check runs against a segment in utility mode and the distribution
policy isn't stored in the segments.

The parentPolicy pointer is used at the make_distributedby_for_rel function.
We should execute this function only if the parentPolicy pointer (
"parentrel->rd_cdbpolicy") is not NULL.

### CODE REVIEW

File: src/backend/parser/parse_utilcmd.c [hunk 1]
```diff
@@ -2031,7 +2031,7 @@ transformDistributedBy(CreateStmtContext *cxt,
 			 * is an inherited table, set the distribution based on the
 			 * parent (or one of the parents)
 			 */
-			if (distrkeys == NIL && parentPolicy->nattrs >= 0)
+			if (distrkeys == NIL && parentPolicy && parentPolicy->nattrs >= 0)
```
Comment (RekGRpth) [inline src/backend/parser/parse_utilcmd.c]:
```suggestion
			if (distrkeys == NIL && parentPolicy != NULL && parentPolicy->nattrs >= 0)
```

**Comment by BenderArenadata** [comment]
Allure report https://allure-ee.adsw.io/launch/63348

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1036716

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1036717

File: src/backend/parser/parse_utilcmd.c (hunk 1 above)
Comment (red1452) [reply src/backend/parser/parse_utilcmd.c]: Is there code style, where there is this requirement?

Comment (RekGRpth) [reply src/backend/parser/parse_utilcmd.c]: > Is there code style

[yes](https://github.com/arenadata/gpdb/pull/700/files#diff-5bd59ecc8991bacaefd56f7fe986287b8d664e62566eb3588c3845d7625cacf1R2018)

Comment (red1452) [reply src/backend/parser/parse_utilcmd.c]: I said about rules, not examples in the code.

**Review by RekGRpth — APPROVED** [review]

**Comment by andr-sokolov** [comment]
@red1452 Explain please how the heap_create_with_catalog() function is connected to transformDistributedBy() or may be remove the mention of heap_create_with_catalog(). Explain why null value in parentPolicy is handled this way.

**Comment by red1452** [comment]
> @red1452 Explain please how the heap_create_with_catalog() function is connected to transformDistributedBy() or may be remove the mention of heap_create_with_catalog(). Explain why null value in parentPolicy is handled this way.

I updated the description.

**Pull request merged**